### PR TITLE
change PhoneNumber and use it in ContactInformation

### PR DESCRIPTION
**Pull request by daschJulia**
only format phonenumber when getInternationalFormatOfPhoneNumber() is called.
do only format if the rawPhoneNumber of the PhoneNumber is not null and no
PhoneNumberFormattingException is thrown.
PhoneNumber is used in ContactInformation instead of Strings of a phone number.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -75,6 +75,11 @@
             <version>1.10.19</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
```
Comment (rjayasinghe) [inline pom.xml]: Uhoh. I guess we'd like to stay without a dependency to Spring framework. If this has been added for the StringUtils use case please use something similar from e.g. apache-commons or guava etc.

File: src/main/java/net/contargo/types/contactinfo/normalization/PhoneNumberNormalizer.java [hunk 2]
```diff
@@ -11,16 +12,10 @@
 
     public Optional<String> normalizeNumber(String phoneNumber) {
 
-        if (phoneNumber == null) {
+        if (StringUtils.hasText(phoneNumber)) {
```
Comment (rjayasinghe) [inline src/main/java/net/contargo/types/contactinfo/normalization/PhoneNumberNormalizer.java]: I guess this is the one. I had a look at the other code in this project. There are cases where we use StringUtils.isEmpty() from Apache Commons. Please adjust your imports. :)

**Comment by rjayasinghe** [comment]
Looks good so far. Please adjust the Spring framework dependency. After that somebody can merge. :)

**Review by rjayasinghe — APPROVED** [review]
Looks good now. :)

File: src/main/java/net/contargo/types/telephony/PhoneNumber.java [hunk 3]
```diff
@@ -58,7 +65,11 @@ public void setPhoneExtension(String phoneExtension) {
 
     public Country getCountry() throws PhoneNumberFormattingException {
 
-        return new Country(phoneNumberFormatter.getRegionCode(rawPhoneNumber));
+        if (Objects.isNull(country)) {
```
Comment (rjayasinghe) [inline src/main/java/net/contargo/types/telephony/PhoneNumber.java:68]: I think a StringUtils.isBlank() would be better here.

Comment (daschJulia) [reply src/main/java/net/contargo/types/telephony/PhoneNumber.java:68]: Country is an Object :)

**Pull request merged**